server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -1216,7 +1218,7 @@ public final int hashCode() {
     }
 
     private enum VectorIndexType {
-        HNSW("hnsw") {
+        HNSW("hnsw", false) {
```

Comment:
I wonder why `hnsw` index doesn't support oversampling? 
On some datasets the performance on hnsw can be bad.

Reply:
Original scope was for quantized vectors. I guess we could expand it to non-quantized as well, WDYT @benwtrent ?

Reply:
> I wonder why hnsw index doesn't support rescoring with oversampling?

@mayya-sharipova there is no reason to rescore `float32`. Why would you rescore floating point values with the exact same indices. 

If somebody wants to "oversample" they should simply increase `num_candidates`.

Reply:
They key thing here is the "rescoring" part.